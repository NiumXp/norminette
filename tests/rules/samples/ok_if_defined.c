#if defined FOO
# undef FOO
#elif defined (BAR)
# undef BAR
#elif defined(FOOBAR)
# undef FOOBAR
#endif /* if defined FOO */

#define A defined B

void	defined(void);
