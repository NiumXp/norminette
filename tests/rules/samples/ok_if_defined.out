[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 1":
		<HASH> <IF> <SPACE> <IDENTIFIER=defined> <SPACE> <IDENTIFIER=FOO> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 2":
		<HASH> <SPACE> <IDENTIFIER=undef> <SPACE> <IDENTIFIER=FOO> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 3":
		<HASH> <IDENTIFIER=elif> <SPACE> <IDENTIFIER=defined> <SPACE> <LPARENTHESIS> <IDENTIFIER=BAR> <RPARENTHESIS> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 4":
		<HASH> <SPACE> <IDENTIFIER=undef> <SPACE> <IDENTIFIER=BAR> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 5":
		<HASH> <IDENTIFIER=elif> <SPACE> <IDENTIFIER=defined> <LPARENTHESIS> <IDENTIFIER=FOOBAR> <RPARENTHESIS> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 6":
		<HASH> <SPACE> <IDENTIFIER=undef> <SPACE> <IDENTIFIER=FOOBAR> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 7":
		<HASH> <IDENTIFIER=endif> <SPACE> <MULT_COMMENT=/* if defined FOO */> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 8":
		<NEWLINE>
[36mok_if_defined.c[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 9":
		<HASH> <IDENTIFIER=define> <SPACE> <IDENTIFIER=A> <SPACE> <IDENTIFIER=defined> <SPACE> <IDENTIFIER=B> <NEWLINE>
[36mok_if_defined.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 10":
		<NEWLINE>
[36mok_if_defined.c[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 11":
		<VOID> <TAB> <IDENTIFIER=defined> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
ok_if_defined.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
Error: PREPROC_CONSTANT     (line:   9, col:  19):	Preprocessor statement must only contain constant defines
